manifests/agent.pp

```diff
@@ -325,6 +339,18 @@
     }
   }
 
+  # If both epp and erb are defined, throw validation error.
+  # Otherwise use the defined erb/epp template, or use default
+  if $agent_configfile_epp and $agent_configfile_template {
+    fail('Cannot supply both agent_configfile_epp and agent_configfile_template templates for config file.')
+  } elsif $agent_configfile_template {
+    $config_content = template($agent_configfile_template)
+  } elsif $agent_configfile_epp {
+    $config_content = epp($agent_configfile_epp)
+  } else {
+    $config_content = template('zabbix/zabbix_agentd.conf.erb')
+  }
```

Comment:
I'm not sure I understand the logic, but I don't think the user should define where to pick the template (either epp or erb). That should be predefined in the module.

Reply:
I used the same logic here as puppetlabs-ntp uses. I can however change it if it's not the preferred way.

Reply:
Selector to the rescue